algorithms/linfa-clustering/src/optics/algorithm.rs

```diff
@@ -109,6 +109,18 @@ pub struct OpticsAnalysis<F> {
     orderings: Vec<Sample<F>>,
 }
 
+impl<F: Float> PartialEq for OpticsAnalysis<F> {
+    fn eq(&self, other: &Self) -> bool {
+        self.orderings == other.orderings
+    }
+}
+
+impl<F: Float> PartialOrd for OpticsAnalysis<F> {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        self.orderings.partial_cmp(&other.orderings)
+    }
+}
+
```

Comment:
You can just derive these traits on `Sample` and on `OpticsAnalysis`

Reply:
I think for `Sample` it won't work as it's only must be done for `reachability_distance`, with the `derive` it will check all parameters AFAIK.

For `OpticsAnalysis` it throws the error `error[E0369]: binary operation '==' cannot be applied to type 'Vec<Sample<F>>'` when using it through `derive`

Reply:
Didn't realize that the impls on `Sample` were already there before. In that case just keep the logic as is. However you can change the bounds of those impls from `F: Float` to `F: <Trait_name>`. This will make the derive on `OpticsAnalysis` work as well.

Reply:
Also delete the `PartialOrd` impl

Reply:
Removed `PartialOrd`.
For `PartialEq` in `OpticsAnalysis` it was sufficient to add bound `F: Float`, as there is impl of `PartialEq` for `Sample<F: Float>`. Just adding `F: PartialEq` wasn't enough